### PR TITLE
test: Update snapshot test to cleanup test resources after each test run

### PR DESCRIPTION
**Pull request by chohongm**
## Changes
- N/A

ticket: [AC-4343]

## Additional Notes
- Updated snapshot test to cleanup test resources after each test run
- You can check test resources being created and then deleted in real time during test run
- [Run test locally](https://sendbird.atlassian.net/wiki/spaces/AC/pages/2610724994/Widget+snapshot+test+plan)

## Checklist
Before requesting a code review, please check the following:
- [x] **[Required]** CI has passed all checks.
- [x] **[Required]** A self-review has been conducted to ensure there are no minor mistakes.
- [x] **[Required]** Unnecessary comments/debugging code have been removed.
- [x] **[Required]** All requirements specified in the ticket have been accurately implemented.
- [ ] Ensure the ticket has been updated with the sprint, status, and story points.

[AC-4343]: https://sendbird.atlassian.net/browse/AC-4343?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: __visual_tests__/const.ts [hunk 1]
```diff
@@ -1,7 +1,16 @@
...
+export const AppId = process.env.SNAPSHOT_TEST_APP_ID;
+export const BotId = process.env.SNAPSHOT_TEST_BOT_ID;
+export const ApiToken = process.env.SNAPSHOT_TEST_API_TOKEN;
 
-export const TEST_URL = `http://localhost:5173/chat-ai-widget/?app_id=${appId}&bot_id=${botId}&snapshot=true`;
+const apiVersion = '/v3';
+export const PlatformApiPath = {
+  GROUP_CHANNELS: `${apiVersion}/group_channels/`,
+  USERS: `${apiVersion}/users/`,
+};
```
Comment (bang9) [inline __visual_tests__/const.ts]: path 는 deleteChannel 와 deleteUser 함수 안에 그냥 들어가는게 더 깔끔할 것 같은데 어떠신가유?

File: __visual_tests__/utils/requestUtils.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    'Api-Token': ApiToken,
+    'Content-Type': 'application/json',
+  };
+}
+
+async function requestDelete(requestParams: RequestParams) {
+  const response = await fetch(
+    `${ApiHost}${requestParams.url}?${createQueryString({
+      ...requestParams.data,
+      sendbird: uuid(),
```
Comment (bang9) [inline __visual_tests__/utils/requestUtils.ts]: uuid 는 혹시 필요한건가용?

Comment (chohongm) [reply __visual_tests__/utils/requestUtils.ts]: 빼고 돌려봤는데도 잘 되는걸로 보아 필요없어 보입니다. 지웠습니다.

File: __visual_tests__/const.ts (hunk 1 above)
Comment (chohongm) [reply __visual_tests__/const.ts]: done

File: __visual_tests__/utils/requestUtils.ts (hunk 2 above)
Comment (chohongm) [reply __visual_tests__/utils/requestUtils.ts]: done

File: __visual_tests__/utils/localStorageUtils.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+export async function getWidgetSessionCache(
+  page: Page,
+  { appId, botId }: { appId: string; botId: string },
+): Promise<WidgetSessionCache | null> {
+  const key = getKey(appId, botId);
+  const value = await page.evaluate(
+    ({ key }) => {
+      return localStorage.getItem(key); // Replace 'key' with your key
+    },
+    { key },
```
Comment (bang9) [inline __visual_tests__/utils/localStorageUtils.ts]: key 변수 스코프가 헷갈려서 아래처럼 수정하면 좋을 것 같아요.

```suggestion
  const value = await page.evaluate(
    ({ key }) => (localStorage.getItem(key)),
    { key: getKey(appId, botId) },
```

File: __visual_tests__/utils/localStorageUtils.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  try {
+    if (value) {
+      // For cache of users before the update, there is no 'strategy'.
+      // Therefore, 'auto' is set as the default value.
+      return { strategy: 'auto', ...JSON.parse(value) };
+    }
+    return null;
+  } catch {
+    return null;
+  }
```
Comment (bang9) [inline __visual_tests__/utils/localStorageUtils.ts]: 이거 startegy auto 없이 그냥 바로 리턴해도 괜찮지 않을까여?

Comment (bang9) [reply __visual_tests__/utils/localStorageUtils.ts]:
```
if (value) {
  try {
    return JSON.parse(value);
  } catch {
    return null;
  }
} else {
  return null;
}
```

File: __visual_tests__/utils/localStorageUtils.ts [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  return `${WIDGET_SESSION_PREFIX}/${appId}/${botId}`;
+};
+
+export type WidgetSessionCache = {
+  strategy: 'auto' | 'manual';
+  userId: string;
+  channelUrl: string;
+  expireAt: number;
+  sessionToken?: string;
+};
```
Comment (bang9) [inline __visual_tests__/utils/localStorageUtils.ts]: 리소스 클린하는 입장에서는 user id, channel url 만 알아도 될 것 같습니다.

File: __visual_tests__/utils/localStorageUtils.ts (hunk 4 above)
Comment (chohongm) [reply __visual_tests__/utils/localStorageUtils.ts]: done!

File: __visual_tests__/utils/localStorageUtils.ts (hunk 5 above)
Comment (chohongm) [reply __visual_tests__/utils/localStorageUtils.ts]: done!

File: __visual_tests__/utils/localStorageUtils.ts [hunk 6]
```diff
@@ -0,0 +1,27 @@
+import { Page } from '@playwright/test';
+
+export const WIDGET_SESSION_PREFIX = '@sendbird/chat-ai-widget';
+export const getKey = (appId: string, botId: string) => {
+  return `${WIDGET_SESSION_PREFIX}/${appId}/${botId}`;
+};
```
Comment (bang9) [inline __visual_tests__/utils/localStorageUtils.ts]:
```suggestion
export const getKey = (appId: string, botId: string) => {
  return `@sendbird/chat-ai-widget/${appId}/${botId}`;
};
```

File: __visual_tests__/utils/requestUtils.ts [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+    'Api-Token': ApiToken,
+    'Content-Type': 'application/json',
+  };
+}
+
+async function requestDelete(requestParams: RequestParams) {
+  const response = await fetch(
+    `${ApiHost}${requestParams.url}?${createQueryString({
+      ...requestParams.data,
+    })}`,
```
Comment (bang9) [inline __visual_tests__/utils/requestUtils.ts]:
```suggestion
    `${ApiHost}${requestParams.url}?${createQueryString(requestParams.data)}`,
```

**Review by bang9 — APPROVED** [review]

**Pull request merged**